### PR TITLE
Refactor ray rdd xshards

### PR DESCRIPTION
**Pull request by yushan111**
This PR removes ray rdd and use ray xshards only.
Ray xshards can improve performance of data transfer from spark to ray.

### CODE REVIEW

**Comment by yangw1234** [comment]
LGTM

**Comment by yushan111** [comment]
jenkins：
http://10.239.47.210:18888/view/ZOO-PR/job/ZOO-PR-Validation/5063/
http://10.239.47.210:18888/job/ZOO-PR-Validation-Spark-2.4/3423/

File: pyzoo/zoo/orca/data/__init__.py [hunk 1]
```diff
@@ -16,3 +16,4 @@
 
 from .shard import SparkXShards, SharedValue, XShards
 from .elastic_search import elastic_search
+from .ray_xshards import RayXShards
```
Comment (jason-dai) [inline pyzoo/zoo/orca/data/__init__.py:19]: will this import ray unconditionally?

File: pyzoo/zoo/orca/data/ray_xshards.py [hunk 2]
```diff
@@ -247,22 +285,22 @@ def from_partition_refs(ip2part_id, part_id2ref):
...
-        return RayRdd._from_spark_rdd_ray_api(rdd)
+    def from_spark_xshards(spark_xshards):
+        return RayXShards._from_spark_xshards_ray_api(spark_xshards)
 
     @staticmethod
-    def _from_spark_rdd_ray_api(rdd):
+    def _from_spark_xshards_ray_api(spark_xshards):
         ray_ctx = RayContext.get()
         address = ray_ctx.redis_address
         password = ray_ctx.redis_password
```
Comment (jason-dai) [inline pyzoo/zoo/orca/data/ray_xshards.py:306]: In below:  `if not (not ray_ctx.is_local and key == f"node:{driver_ip}")` --> `if ray_ctx.is_local or key != f"node:{driver_ip}")`

**Comment by yushan111** [comment]
fix new comments in PR #3500

**Pull request merged**